File: src/test/java/com/github/alexeysol/geekregimeapiposts/TestUtils.java
package com.github.alexeysol.geekregimeapiposts;

import com.fasterxml.jackson.databind.ObjectMapper;
import org.springframework.http.MediaType;
import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder;
import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;

public class TestUtils {
    static private final ObjectMapper objectMapper = new ObjectMapper();

    static public <Value> String objectToJsonString(Value obj) {
        try {
            return objectMapper.writeValueAsString(obj);
        } catch (Exception exception) {
            throw new RuntimeException(exception);
        }
    }

    static public <Value> MockHttpServletRequestBuilder mockPostRequest(
        String apiPath,
        Value value
    ) {
        return MockMvcRequestBuilders.post(apiPath)
            .contentType(MediaType.APPLICATION_JSON)
            .content(objectToJsonString(value));
    }

    static public <Value> MockHttpServletRequestBuilder mockPatchRequest(
        String apiPath,
        Value value
    ) {
        return MockMvcRequestBuilders.patch(apiPath)
            .contentType(MediaType.APPLICATION_JSON)
            .content(objectToJsonString(value));
    }
}

File: src/test/kotlin/com/github/alexeysol/geekregimeapiusers/testutils.kt
package com.github.alexeysol.geekregimeapiusers

import com.fasterxml.jackson.databind.ObjectMapper
import org.springframework.http.MediaType
import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder
import org.springframework.test.web.servlet.request.MockMvcRequestBuilders

val objectMapper = ObjectMapper()

fun <Value>objectToJsonString(obj: Value? = null): String {
    try {
        return objectMapper.writeValueAsString(obj);
    } catch (exception: Exception) {
        throw RuntimeException(exception);
    }
}

fun <Value>mockPostRequest(apiPath: String, value: Value? = null): MockHttpServletRequestBuilder {
    return MockMvcRequestBuilders.post(apiPath)
        .contentType(MediaType.APPLICATION_JSON)
        .content(objectToJsonString(value))
}

fun <Item>isSameContentsInArrays(array1: Array<Item>, array2: Array<Item>): Boolean {
    if (array1.size != array2.size) {
        return false
    }

    array1.forEachIndexed { index, item ->
        if (item != array2[index]) {
            return false
        }
    }

    return true
}
